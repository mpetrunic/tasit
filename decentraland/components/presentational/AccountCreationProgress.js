import React from "react";
import { StyleSheet, View, Text } from "react-native";
import PropTypes from "prop-types";
import AccountCreationStatus from "@constants/AccountCreationStatus";
import LinkToEtherscan from "./LinkToEtherscan";
const {
  NOT_STARTED,
  GENERATING_ACCOUNT,
  FUNDING_WITH_ETH,
  FUNDING_WITH_MANA_AND_APPROVING_MARKETPLACE,
  FUNDING_WITH_MANA,
  APPROVING_MARKETPLACE,
  READY_TO_USE,
} = AccountCreationStatus;

export default function AccountCreationProgress(props) {
  const { status, currentAction } = props;

  const waitingForAccountSetup =
    status !== NOT_STARTED && status !== READY_TO_USE;

  let waitingMessage = "";
  switch (status) {
    case NOT_STARTED:
      break;
    case GENERATING_ACCOUNT:
      waitingMessage = "Generating account...";
      break;
    case FUNDING_WITH_ETH:
      waitingMessage = "Funding account with ETH...";
      break;
    case FUNDING_WITH_MANA_AND_APPROVING_MARKETPLACE:
      waitingMessage = "Funding account with MANA and approving marketplace...";
      break;
    case FUNDING_WITH_MANA:
      waitingMessage = "Funding account with MANA...";
      break;
    case APPROVING_MARKETPLACE:
      waitingMessage = "Approving marketplace...";
      break;
    case READY_TO_USE:
      break;
    default:
  }

  if (!waitingForAccountSetup) return null;

  return (
    <View style={styles.textRow}>
      <Text>{waitingMessage}</Text>
      <LinkToEtherscan action={currentAction} />
    </View>
  );
}

AccountCreationProgress.propTypes = {
  status: PropTypes.string.isRequired,
  currentAction: PropTypes.object,
};

const styles = StyleSheet.create({
  textRow: { flexDirection: "row" },
});
