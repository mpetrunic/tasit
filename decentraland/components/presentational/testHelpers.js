import AssetTypes from "@constants/AssetTypes";
const { ESTATE, PARCEL } = AssetTypes;

export const parcel = {
  type: PARCEL,
  id: "0123456789",
  name: "Sample Parcel",
  img: "https://api.decentraland.org/v1/parcels/0/0/map.png",
};

export const estate = {
  type: ESTATE,
  id: "123",
  name: "Sample Estate",
  img: "https://api.decentraland.org/v1/estates/1/map.png",
};

export const estateForSale = {
  id: "1",
  priceMana: 0,
  priceUSD: 0,
  asset: estate,
};

export const parcelForSale = {
  id: "2",
  priceMana: 0,
  priceUSD: 0,
  asset: parcel,
};

export default {
  estateForSale,
  parcelForSale,
  parcel,
  estate,
};
