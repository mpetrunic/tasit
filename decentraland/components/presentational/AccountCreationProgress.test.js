import React from "react";
import { shallow } from "enzyme";
import AccountCreationProgress from "./AccountCreationProgress";
import status from "@constants/AccountCreationStatus";
const {
  NOT_STARTED,
  GENERATING_ACCOUNT,
  FUNDING_WITH_ETH,
  FUNDING_WITH_MANA_AND_APPROVING_MARKETPLACE,
  FUNDING_WITH_MANA,
  APPROVING_MARKETPLACE,
  READY_TO_USE,
} = status;
import { anAction } from "@helpers/testHelpers";

describe("AccountCreationProgress", () => {
  it("initial state - before start account setup", async () => {
    const status = NOT_STARTED;
    const actions = {};

    expect(
      shallow(<AccountCreationProgress status={status} actions={actions} />)
    ).toMatchSnapshot();
  });

  it("waiting for setup - waiting for account generation", async () => {
    const status = GENERATING_ACCOUNT;
    const actions = {};

    expect(
      shallow(<AccountCreationProgress status={status} actions={actions} />)
    ).toMatchSnapshot();
  });

  it("waiting for setup - waiting for ETH funding", async () => {
    const status = FUNDING_WITH_ETH;
    const actions = { FUNDING_WITH_ETH: anAction };

    expect(
      shallow(<AccountCreationProgress status={status} actions={actions} />)
    ).toMatchSnapshot();
  });

  it("waiting for setup - waiting for MANA funding and marketplace approval", async () => {
    const status = FUNDING_WITH_MANA_AND_APPROVING_MARKETPLACE;
    const actions = { FUNDING_WITH_MANA_AND_APPROVING_MARKETPLACE: anAction };

    expect(
      shallow(<AccountCreationProgress status={status} actions={actions} />)
    ).toMatchSnapshot();
  });

  it("waiting for setup - waiting for MANA funding", async () => {
    const status = FUNDING_WITH_MANA;
    const actions = { FUNDING_WITH_MANA: anAction };

    expect(
      shallow(<AccountCreationProgress status={status} actions={actions} />)
    ).toMatchSnapshot();
  });

  it("waiting for setup - waiting for marketplace approval", async () => {
    const status = APPROVING_MARKETPLACE;
    const actions = { APPROVING_MARKETPLACE: anAction };

    expect(
      shallow(<AccountCreationProgress status={status} actions={actions} />)
    ).toMatchSnapshot();
  });

  it("final state - account ready", async () => {
    const status = READY_TO_USE;
    const actions = {};

    expect(
      shallow(<AccountCreationProgress status={status} actions={actions} />)
    ).toMatchSnapshot();
  });
});
