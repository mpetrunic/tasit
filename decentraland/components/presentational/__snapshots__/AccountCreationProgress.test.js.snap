// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountCreationProgress final state - account ready 1`] = `""`;

exports[`AccountCreationProgress initial state - before start account setup 1`] = `""`;

exports[`AccountCreationProgress waiting for setup - waiting for ETH funding 1`] = `
<Component
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <Component>
    Funding account with ETH...
  </Component>
  <LinkToEtherscan />
</Component>
`;

exports[`AccountCreationProgress waiting for setup - waiting for MANA funding 1`] = `
<Component
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <Component>
    Funding account with MANA...
  </Component>
  <LinkToEtherscan />
</Component>
`;

exports[`AccountCreationProgress waiting for setup - waiting for MANA funding and marketplace approval 1`] = `
<Component
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <Component>
    Funding account with MANA and approving marketplace...
  </Component>
  <LinkToEtherscan />
</Component>
`;

exports[`AccountCreationProgress waiting for setup - waiting for account generation 1`] = `
<Component
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <Component>
    Generating account...
  </Component>
  <LinkToEtherscan />
</Component>
`;

exports[`AccountCreationProgress waiting for setup - waiting for marketplace approval 1`] = `
<Component
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <Component>
    Approving marketplace...
  </Component>
  <LinkToEtherscan />
</Component>
`;
